src/test/java/de/tum/in/www1/artemis/exercise/programmingexercise/ProgrammingSubmissionAndResultBitbucketBambooIntegrationTest.java

```diff
@@ -410,6 +415,9 @@ void shouldNotCreateTwoSubmissionsForTwoIdenticalCommits(IntegrationTestParticip
         doReturn(defaultBranch).when(versionControlService).getOrRetrieveBranchOfExercise(exercise);
 
         Long participationId = getParticipationIdByType(participationType, 0);
+        // trigger twice because we submit twice
```

Comment:
We submit twice? 🤔 Do you mean in the tests here? Can you add the `mockTrigger...` methods directly after they are needed or is there something preventing that?

Reply:
There are two calls to `postSubmission` underneath; the triggers are mocked before the calls because the `postSubmission` function directly calls the API, which triggers a build. I also have to mock them before the calls because adding more mocks is impossible after requests have been made.